File: 3114-latest-time-you-can-obtain-after-replacing-characters/3114-latest-time-you-can-obtain-after-replacing-characters.cs
public class Solution {
    public string FindLatestTime(string s) {
        var ans = s.ToCharArray();
        if(s[0] == '?')
            ans[0] = (s[1] == '?' || s[1] < '2') ? '1' : '0';
        if(s[1] == '?')
            ans[1] = (ans[0] == '1') ? '1' : '9';
        if(s[3] == '?')
            ans[3] = '5';
        if(s[4] == '?')
            ans[4] = '9';

        return new string(ans);
    }
}

File: 3114-latest-time-you-can-obtain-after-replacing-characters/3114-latest-time-you-can-obtain-after-replacing-characters.py
class Solution:
    def findLatestTime(self, s: str) -> str:
        ans = list(s)

        if ans[0] == '?':
            ans[0] = '1' if ans[1] == '?' or ans[1] < '2' else '0'
        if ans[1] == '?':
            ans[1] = '1' if ans[0] == '1' else '9'
        if ans[3] == '?':
            ans[3] = '5'
        if ans[4] == '?':
            ans[4] = '9'

        return ''.join(ans)